macaw-riscv/src/Data/Macaw/RISCV/RISCVReg.hs

```diff
@@ -0,0 +1,451 @@
+{-# LANGUAGE ConstraintKinds #-}
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE PatternSynonyms #-}
+{-# LANGUAGE PolyKinds #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE TypeOperators #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+module Data.Macaw.RISCV.RISCVReg
+  ( -- * RISC-V macaw register state
+    RISCVReg(..)
+    -- ** Patterns for GPRs
+  , pattern GPR
+  , pattern GPR_RA
+  , pattern GPR_SP
+  , pattern GPR_GP
+  , pattern GPR_TP
+  , pattern GPR_T0
+  , pattern GPR_T1
+  , pattern GPR_T2
+  , pattern GPR_S0
+  , pattern GPR_S1
+  , pattern GPR_A0
+  , pattern GPR_A1
+  , pattern GPR_A2
+  , pattern GPR_A3
+  , pattern GPR_A4
+  , pattern GPR_A5
+  , pattern GPR_A6
+  , pattern GPR_A7
+  , pattern GPR_S2
+  , pattern GPR_S3
+  , pattern GPR_S4
+  , pattern GPR_S5
+  , pattern GPR_S6
+  , pattern GPR_S7
+  , pattern GPR_S8
+  , pattern GPR_S9
+  , pattern GPR_S10
+  , pattern GPR_S11
+  , pattern GPR_T3
+  , pattern GPR_T4
+  , pattern GPR_T5
+  , pattern GPR_T6
+  ) where
+
+import qualified Data.Macaw.Types as MT
+import qualified Data.Macaw.CFG as MC
+
+import           Data.Macaw.RISCV.Arch
+
+import           Data.Parameterized.Classes
+import           Data.Parameterized.Some (Some(..))
+import           Data.Parameterized.NatRepr (knownNat)
+import           Data.Parameterized.TH.GADT ( structuralTypeEquality
+                                            , structuralTypeOrd
+                                            )
+
+import qualified GRIFT.Types as G
+import qualified GRIFT.Semantics.Utils as G
+
+newtype GPR = BuildGPR (G.SizedBV 5)
+  deriving (Enum, Eq, Num, Ord)
```

Comment:
Is this `Num` instance used? It would be nice to not have it if not (to avoid invalid register creation from literals)

Reply:
Nope!  I removed it.